File: dataset_reduzido/codenet/Java/Code/s537034016.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	static long[] dp;
	static int mod = 1000000007;

	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);
		int s = sc.nextInt();
		sc.close();

		dp = new long[s + 1];
		Arrays.fill(dp, -1);
		dp[0] = 1;
		System.out.println(dfs(s));
	}

	static long dfs(int s) {
		if (dp[s] != -1) {
			return dp[s];
		}

		long ret = 0;
		for (int i = s - 3; i >= 0; i--) {
			ret += dfs(i);
			ret %= mod;
		}
		return dp[s] = ret;
	}
}

File: output/Qwen/codenet/Java/Python/s537034016.py
def dfs(s, dp, mod):
    if dp[s] != -1:
        return dp[s]
    
    ret = 0
    for i in range(s - 3, 0, -1):
        ret += dfs(i, dp, mod)
        ret %= mod
    return dp[s] = ret

s = int(input())
mod = 1000000007
dp = [-1] * (s + 1)
dp[0] = 1

print(dfs(s, dp, mod))